engine/src/main/java/org/terasology/logback/RegexFilterAction.java

```diff
@@ -0,0 +1,180 @@
+import ch.qos.logback.core.joran.action.Action;
+import ch.qos.logback.core.joran.spi.ActionException;
+import ch.qos.logback.core.joran.spi.InterpretationContext;
+import ch.qos.logback.core.spi.FilterAttachable;
+import ch.qos.logback.core.spi.FilterReply;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Strings;
+import org.xml.sax.Attributes;
+
+import java.util.function.BiFunction;
+
+/**
+ * Regex-based logback filter with concise syntax.
+ * <p>
+ * <b>Example:</b>
+ * <p><code>
+ *     &lt;denyRegex prefix="org.reflections" message="given scan \w+ are empty" /&gt;
+ * </code>
+ * <dl>
+ *     <dt>{@code denyRegex}</dt>
+ *     <dd>The tag starts with {@code "deny"}, meaning events matching this filter will be dropped.</dd>
+ *
+ *     <dt>{@code prefix} <i>(optional)</i></dt>
+ *     <dd>This filter will only test events from loggers with names starting {@code "org.reflections"}.<br>
+ *     It is {@linkplain FilterReply#NEUTRAL neutral} on other events.</dd>
+ *
+ *     <dt>{@code message}</dt>
+ *     <dd>Match the {@linkplain ILoggingEvent#getFormattedMessage() formatted message} against
+ *     the regular expression {@linkplain java.util.regex.Pattern pattern}
+ *     {@code /given scan \w+ are empty/}.</dd>
+ * </dl>
+ * <p>
+ * <b>Example:</b>
+ * <p><code>
+ *     &lt;requireRegex message="module:.*-SNAPSHOT" /&gt;
+ * </code>
+ * <dl>
+ *     <dt>{@code requireRegex}</dt>
+ *     <dd>The tag starts with {@code "require"}, meaning events <em>must</em> match the pattern.</dd>
+ *
+ *     <dt>{@code message}</dt>
+ *     <dd>Match the {@linkplain ILoggingEvent#getFormattedMessage() formatted message} against
+ *     the regular expression {@linkplain java.util.regex.Pattern pattern}
+ *     {@code /module:.*-SNAPSHOT/}.</dd>
+ * </dl>
+ * No Groovy or Janino required.
+ * <p>
+ * This Action adds an {@link EvaluatorFilter} with a {@link RegexEvaluator} to
+ * the current element, such as an {@link ch.qos.logback.core.Appender Appender}.
+ * <p>
+ * The element must be on the top of the stack and must implement {@link FilterAttachable}.
+ * <p>
+ * <b>Implementation Note:</b> In logback 1.2, we can <em>almost</em> define an
+ * EvaluatorFilter using only logback-core, but core defines no LoggingEvent interface.
+ * We have to link against logback-classic for that.
+ *
+ * @param <E> the type of logging event the filter operates on
+ */
+public class RegexFilterAction<E extends ILoggingEvent> extends Action {
+    public static final String PREFIX_ATTRIBUTE = "prefix";
+    public static final String MESSAGE_ATTRIBUTE = "message";
+    public static final String DENY_ELEMENT = "deny";
+    public static final String REQUIRE_ELEMENT = "require";
+
+    protected final BiFunction<String, String, EventEvaluator<E>> evaluatorConstructor = RegexEvaluator::new;
+
+    @Override
+    public void begin(InterpretationContext ic, String elementName, Attributes attributes) throws ActionException {
```

Comment:
```suggestion
    public void begin(InterpretationContext context, String elementName, Attributes attributes) throws ActionException {
```

can we call this `context`? I don't think it's necessary to make the name as small as possible and it might help with reading the code.

Reply:
Can do.

For `@Override` methods like this that are implementing the interface, I probably defaulted to the spelling used in the interface's declaration, but changing the parameter names doesn't hurt.